src/chai/ManagedArray.inl

```diff
@@ -324,6 +338,11 @@ CHAI_INLINE
 CHAI_HOST
 void ManagedArray<T>::move(ExecutionSpace space)
 {
+  /* If this ManagedArray isn't associated with an allocation return immediately. */
+  if (m_pointer_record == nullptr) {
```

Comment:
We have nested `ManagedArrays` where some inner arrays are empty and others are assigned to using the shallow copy `operator=`. We initialize these inner arrays using the `ManagedArray(nullptr_t)` constructor since the default constructor allocates a `PointerRecord` that needs to be free'd. When you copy-construct or move the outer array it moves the inner arrays, some of which have not been assigned to (`m_pointer_record == nullptr`).